Code-the-Dream-School/dd-prac-team4-front
# PS-108 Create a react component for the wishlist button

**Pull Request Description:**

**Title:** Implementing Add to Wishlist Button Component

**Description:**
This pull request introduces the "AddToWishlistButton" component to the application. This component provides the functionality for users to add or remove albums to/from their wishlists. The button's behavior changes based on the user's login status and whether the album has already been added to the wishlist.

**Changes Made:**
- Added a new component named "AddToWishlistButton" responsible for managing the addition and removal of albums from the wishlist.
- The "AddToWishlistButton" component provides a seamless way for users to interact with their wishlists, enhancing their experience by allowing them to manage their favorite albums effortlessly.
- Users can seamlessly add or remove albums from their wishlist using the same button. The button intelligently switches between "Add to Wishlist" and "Remove from Wishlist" based on the album's current wishlist status, streamlining the interaction process.
- Users are directed to the sign-in page if they are not logged in and attempt to add an album to the wishlist.

**Related Back-end Pull Request**
https://github.com/Code-the-Dream-School/dd-prac-team4-back/pull/42

**Testing steps**

1. Click the button to either "Add to Wishlist" or "Remove from Wishlist."
2. After each button click, examine the wishlist collection in the database.
3. Verify that the album's status in the wishlist corresponds to the action performed.
4. Validate that the database accurately reflects changes based on the button's toggle functionality.
5. Click the button while logged out and observe whether you're redirected to the login page.

**Screenshots:**
<img width="263" alt="Screenshot 2023-08-30 at 12 54 29 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/2952900/2d843a98-add4-4d1b-97a3-e180d9368f39">
<img width="298" alt="Screenshot 2023-08-30 at 1 04 05 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/2952900/f639ddc0-a5d4-45a2-b2d6-6bf74aca3e1e">
